[DOCS] Osquery features in 8.5 (#2561)
elastic/security-docs: docs/detections/osquery-response-action.asciidoc

[[osquery-response-action]]
=== Add Osquery Response Actions
Osquery Response Actions allow you to add live queries to custom query rules so you can automatically collect data on systems the rule is monitoring. Use this data to support your alert triage and investigation efforts.

> NOTE: Response Actions are in technical preview.

Author: ```suggestion
preview::[]
```

[role="screenshot"]

> image::images/available-response-actions.png[Shows available response actions]

Reviewer: "Shows" is unnecessary for alt text; every image shows something, so we don't need to say it every time. Most alt text guidelines I've seen (Elastic's [accessibility guidelines](https://docs.elastic.dev/content-design/accessibility#guidelines) and other [industry guidelines](https://webaim.org/techniques/alttext/)) just recommend simple descriptions, no verbs.

> image::images/available-response-actions.png[Shows available response actions]

Author: ```suggestion
image::images/available-response-actions.png[Available response actions]
```

> .Requirements

Reviewer: Maybe move the reqs above the screenshot -- they kind of get buried under the image?

[sidebar]
--
* Osquery Response Actions are available with a https://www.elastic.co/pricing[Platinum or Enterprise subscription].
* The {kibana-ref}/manage-osquery-integration.html[Osquery manager integration] must be installed.
* {agent}'s {fleet-guide}/view-elastic-agent-status.html[status] must be `Healthy`. Refer to {fleet-guide}/fleet-troubleshooting.html[{fleet} Troubleshooting] if it isn't.
* Your role must have {kibana-ref}/osquery.html[Osquery feature privileges].
--

[float]
[[add-osquery-response-action]]
=== Add Osquery Response Actions to rules

> You can add Osquery Response Actions to new or existing custom query rules. Queries are ran every time the rule executes.

Reviewer: Or just "Queries run every time" to avoid passive voice
```suggestion
You can add Osquery Response Actions to new or existing custom query rules. Queries are run every time the rule executes.
```

. Choose one of the following:
** *New rule*: When you are on the last step of <<create-custom-rule,custom query rule>>, go to the Response Actions section and click the *osquery* icon.
** *Existing rule*: Edit the rule's settings, then go to the *Actions* tab. In the tab, click the *osquery* icon under the Response Actions section.
. Specify whether you want to set up a single live query or a pack:

> ** *Query*: Select a saved query or enter a new one in the text box. After you enter the query, you can expand the **Advanced** section to view or set {kibana-ref}/osquery.html#osquery-map-fields[mapped ECS fields] included in the results from the live query. Mapping ECS fields is optional.

Reviewer: Seems clear where the user can enter a query
```suggestion
** *Query*: Select a saved query or enter a new one. After you enter the query, you can expand the **Advanced** section to view or set {kibana-ref}/osquery.html#osquery-map-fields[mapped ECS fields] included in the results from the live query. Mapping ECS fields is optional.
```

** *Pack*: Select from query packs that have been loaded and activated. After you select a pack, all of the queries in the pack are displayed.
+
TIP: Refer to {kibana-ref}/osquery.html#osquery-prebuilt-packs-queries[prebuilt packs] to learn about using and managing Elastic prebuilt packs.
+
[role="screenshot"]
image::images/setup-single-query.png[Shows how to set up a single query]
+

. Click the *osquery* icon to add more live queries (optional).

> . Create the rule or save your changes to finish adding the queries.

Reviewer: Unclear what's the difference between "create the rule" and "save your changes" -- at this point in creating/editing a rule, the only thing the user can do to proceed is click "Save changes".

Author: @joepeeples at the core, there's not a difference between creating a new rule and saving changes to an existing one that you're updating. That said, I did want to list both options because the action is slightly different in the UI. For example when creating a new rule, users would click **Create & enable rule** (or **Create rule without enabling it**) to save the rule. When editing, an existing rule, they'd save their changes by clicking **Save changes**.

Reviewer: @nastasha-solomon Gotcha, thanks for explaining. The rest of the procedure is nicely specific about what the user should do, click, etc., and this step seemed less clear on what exactly to do. Would this work?
```suggestion
. Click **Create & enable rule** (for a new rule) or **Save changes** (for existing rules) to finish adding the queries.
```

[float]
[[edit-osquery-response-action]]
=== Edit Osquery Response Actions

If you want to choose a different query or query pack for the Osquery Response Action to use, edit the rule to update the Response Action.

IMPORTANT: If you edited a saved query or query pack that an Osquery Response Action is using, you must reselect the saved query or query pack on the related Osquery Response Action. Query changes are not automatically applied to Osquery Response Actions.

. Edit the rule's settings, then go to the *Actions* tab.
. Modify the settings for Osquery Response Actions you've added.
. Click *Save changes*.

[float]
[[find-osquery-response-action-results]]
=== Find query results

> When an alert is generated, Osquery automatically collects data on the system related to the alert. Query results are displayed within the *Osquery Results* tab in the Alert details flyout. The number to the right of the *Osquery Results* tab represents the number of queries attached to the rule.

Reviewer: ```suggestion
When an alert is generated, Osquery automatically collects data on the system related to the alert. Query results are displayed within the *Osquery Results* tab in the Alert details flyout. The number next to the *Osquery Results* tab represents the number of queries attached to the rule.
```
